### PR TITLE
Remove remaining devdocs and cleanup ComponentModel code

### PR DESCRIPTION
**Pull request by hughbe**

### CODE REVIEW

File: src/System.Private.CoreLib/shared/System/ComponentModel/DefaultValueAttribute.cs [hunk 1]
```diff
@@ -80,173 +77,162 @@ bool TryConvertFromInvariantString(Type typeToConvert, string stringValue, out o
...
 
         public override bool Equals(object obj)
         {
             if (obj == this)
             {
                 return true;
             }
-
-            if (obj is DefaultValueAttribute other)
+            if (!obj is DefaultValueAttribute other)
```
Comment (stephentoub) [inline src/System.Private.CoreLib/shared/System/ComponentModel/DefaultValueAttribute.cs]: You need parens around the `obj is DefaultValueAttribute other`; otherwise this won't compile.

File: src/System.Private.CoreLib/shared/System/ComponentModel/EditorBrowsableAttribute.cs [hunk 2]
```diff
@@ -7,40 +7,31 @@ namespace System.ComponentModel
...
+                return false;
+            }
 
-            return (obj is EditorBrowsableAttribute other) && other.browsableState == browsableState;
-        }
-
-        public override int GetHashCode()
-        {
-            return base.GetHashCode();
+            return other.State == State;
```
Comment (stephentoub) [inline src/System.Private.CoreLib/shared/System/ComponentModel/EditorBrowsableAttribute.cs]: Why did you change this from what it was?  That's a very common form for writing out such statements.

File: src/System.Private.CoreLib/shared/System/ComponentModel/DefaultValueAttribute.cs (hunk 1 above)
Comment (hughbe) [reply src/System.Private.CoreLib/shared/System/ComponentModel/DefaultValueAttribute.cs]: Derp I swear I compiled - must not have saved

**Comment by stephentoub** [comment]
@dotnet-bot test Ubuntu x64 Checked CoreFX Tests please
@dotnet-bot test Windows_NT x64 Checked CoreFX Tests please
@dotnet-bot test Windows_NT x86 Release Innerloop Build and Test please

**Comment by hughbe** [comment]
@dotnet-bot Test Windows_NT x86 Release Innerloop Build and Test
@dotnet-bot Test coreclr-ci (Build Linux arm64 checked)

**Comment by hughbe** [comment]
Good to merge?
